user_guide_src/source/libraries/validation.rst

```diff
@@ -971,6 +971,9 @@ Rule                    Parameter  Description
 ======================= ========== ============================================= ===================================================
 uploaded                Yes         Fails if the name of the parameter does not  ``uploaded[field_name]``
                                     match the name of any uploaded files.
+                                    Prevents form from validating if file upload
+                                    is required.
+
```

Comment:
The description already provides the perfect explanation.

> The validation will fail if the name of the parameter does not match the name of any uploaded files.

Reply:
I don't get what the added sentence means.

Reply:
How about adding this?

> If you want the file upload to be optional (not required), do not define this rule.

Reply:
> How about adding this?
> 
> > If you want the file upload to be optional (not required), do not define this rule.

Yep, that is better, thanks.

Reply:
> The description already provides the perfect explanation.
> 
> > The validation will fail if the name of the parameter does not match the name of any uploaded files.

Technically you must be correct, but there must be a more user-friendly way to say "this rule makes the file required". I read that section and did not get it.

Reply:
Yes. https://github.com/codeigniter4/CodeIgniter4/pull/7969#discussion_r1335243837 is better.